crates/holochain/src/conductor/handle.rs

```diff
@@ -696,8 +696,15 @@ impl<DS: DnaStore + 'static> ConductorHandleImpl<DS> {
         // space retries joining all cells every 5 minutes.
         futures::stream::iter(networks)
             .for_each_concurrent(100, |mut network| async move {
-                if let Err(e) = network.join().await {
-                    tracing::info!(failed_to_join_network = ?e);
+                match tokio::time::timeout(std::time::Duration::from_secs(20), network.join()).await
```

Comment:
[KitsuneTimeout](https://github.com/holochain/holochain/blob/develop/crates/kitsune_p2p/types/src/timeout.rs) is available in develop if you want to use it. Also, should we make this a const or tuning_param?

Reply:
I tried to use KitsuneTimeout but it only allows futures that return a kitsune result.
I made this a generic but I don't know how to put it in the tuning params because I don't see how you can get hold of the params from the network cell handle?